Classes/PluginBase/PluginBase.php

```diff
@@ -160,11 +160,10 @@ protected function baseWrap($content) {
 	 * @param array $configuration configuration array as provided by the TYPO3 core
 	 */
 	protected function initialize($configuration) {
-		$this->conf = $configuration;
-
-		$this->conf = t3lib_div::array_merge_recursive_overrule(
-			$GLOBALS['TSFE']->tmpl->setup['plugin.']['tx_solr.'],
-			$this->conf
+		$this->conf = $GLOBALS['TSFE']->tmpl->setup['plugin.']['tx_solr.'];
+		\TYPO3\CMS\Core\Utility\ArrayUtility::mergeRecursiveWithOverrule(
```

Comment:
Because mergeRecursiveWithOverrule is now passed by reference.
Before it returned the result.
Which means if I want the same result as before, I have $this->conf be $GLOBALS['TSFE']->tmpl->setup['plugin.']['tx_solr.']

otherwise $GLOBALS['TSFE']->tmpl->setup['plugin.']['tx_solr.'] would get the result.